<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Fast tool for programing</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>771</width>
      <height>491</height>
     </rect>
    </property>
    <property name="toolTip">
     <string/>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LayoutDirection::LeftToRight</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab64">
     <attribute name="title">
      <string>Base64</string>
     </attribute>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>711</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Convert Base64 to images and vice versa</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>43</y>
        <width>571</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="lblCaptionImgObtained">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>60</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Generated/Obtained image:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblImgB64">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>81</y>
        <width>391</width>
        <height>336</height>
       </rect>
      </property>
      <property name="tabletTracking">
       <bool>false</bool>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::ContextMenuPolicy::CustomContextMenu</enum>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="text">
       <string>...</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="btnGenImageFromB64">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>138</x>
        <y>420</y>
        <width>101</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>generate image</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnObtainB64">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>420</y>
        <width>111</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>obtain B64 text</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnObtainImg">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>420</y>
        <width>101</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>obtain image</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnGenB64">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>460</x>
        <y>420</y>
        <width>79</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>generateB64</string>
      </property>
     </widget>
     <widget class="QPlainTextEdit" name="txtB64">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>83</y>
        <width>311</width>
        <height>331</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>Paste your Base64 code here</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabguid">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <attribute name="title">
      <string>GUID generator</string>
     </attribute>
     <widget class="Line" name="line_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>571</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="lblCaptionOpenTextFile_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Generator of passwords</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabpass">
     <attribute name="title">
      <string>PASS generator</string>
     </attribute>
     <widget class="QLabel" name="lblCaptionOpenTextFile_2">
      <property name="geometry">
       <rect>
        <x>12</x>
        <y>20</y>
        <width>711</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Generator of passwords</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
      </property>
     </widget>
     <widget class="Line" name="line_3">
      <property name="geometry">
       <rect>
        <x>12</x>
        <y>40</y>
        <width>571</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QCheckBox" name="chbxSymbols">
      <property name="geometry">
       <rect>
        <x>22</x>
        <y>102</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>check to add symbols to pwd</string>
      </property>
      <property name="text">
       <string>symbols</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="chbxNumbers">
      <property name="geometry">
       <rect>
        <x>22</x>
        <y>132</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>check to add numbers to pwd</string>
      </property>
      <property name="text">
       <string>numbers</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="chbxCapitalLetter">
      <property name="geometry">
       <rect>
        <x>22</x>
        <y>162</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>check to add capital letters to pwd</string>
      </property>
      <property name="text">
       <string>capital letters</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinPass">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>73</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>max size is 25</string>
      </property>
      <property name="prefix">
       <string/>
      </property>
      <property name="minimum">
       <number>10</number>
      </property>
      <property name="maximum">
       <number>25</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="stepType">
       <enum>QAbstractSpinBox::StepType::DefaultStepType</enum>
      </property>
      <property name="value">
       <number>15</number>
      </property>
      <property name="displayIntegerBase">
       <number>10</number>
      </property>
     </widget>
     <widget class="QLabel" name="lblCaptionOpenTextFile_5">
      <property name="geometry">
       <rect>
        <x>22</x>
        <y>73</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>width</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="edtGeneratedPass">
      <property name="geometry">
       <rect>
        <x>22</x>
        <y>310</y>
        <width>251</width>
        <height>21</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="lblCaptionOpenTextFile_6">
      <property name="geometry">
       <rect>
        <x>22</x>
        <y>290</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Generated PASS:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
      </property>
     </widget>
     <widget class="QPushButton" name="btnGenPass">
      <property name="geometry">
       <rect>
        <x>22</x>
        <y>200</y>
        <width>79</width>
        <height>24</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>generate a password following the criteria above.</string>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnCopyPass">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>340</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>copy to clipboard</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabhash">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <attribute name="title">
      <string>HASH generator</string>
     </attribute>
     <widget class="QLabel" name="lblCaptionOpenTextFile_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>391</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Generator of hashes</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
      </property>
     </widget>
     <widget class="Line" name="line_5">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>571</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QListWidget" name="lstCryptoMode">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>91</width>
        <height>81</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>SHA256</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>SHA384</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>SHA512</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>MD5</string>
       </property>
      </item>
     </widget>
     <widget class="QPushButton" name="btnGenCrypto">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>310</y>
        <width>79</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Encrypt</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="edtHashGenerated">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>310</y>
        <width>471</width>
        <height>21</height>
       </rect>
      </property>
     </widget>
     <widget class="QCheckBox" name="chbxIsFile">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>190</y>
        <width>121</width>
        <height>20</height>
       </rect>
      </property>
      <property name="toolTip">
       <string/>
      </property>
      <property name="text">
       <string> get from a file</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnGetFileForCrypto">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>220</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>Select a file to generate the hash.</string>
      </property>
      <property name="text">
       <string>get from a file</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="edtOriginEncrypt">
      <property name="geometry">
       <rect>
        <x>147</x>
        <y>221</y>
        <width>431</width>
        <height>21</height>
       </rect>
      </property>
     </widget>
     <widget class="Line" name="line_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>280</y>
        <width>571</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>141</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Select a encrypt mode:</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabfile">
     <attribute name="title">
      <string>Open text file</string>
     </attribute>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>571</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="lblCaptionOpenTextFile">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>711</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Open and search in text files.</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
      </property>
     </widget>
     <widget class="QTextBrowser" name="textFile">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>80</y>
        <width>721</width>
        <height>341</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="btnOpenFileText">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>430</y>
        <width>101</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>open text file</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnSearchInFile">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>430</y>
        <width>79</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>search in file</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lnEdSearchInFile">
      <property name="geometry">
       <rect>
        <x>219</x>
        <y>432</y>
        <width>261</width>
        <height>20</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>input your search text here and press ENTER</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>33</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
